hbase-hbtop/src/main/java/org/apache/hadoop/hbase/hbtop/screen/field/FieldScreenView.java

```diff
@@ -0,0 +1,185 @@
+      return this;
+    }
+
+    assert keyPress.getCharacter() != null;
+    switch (keyPress.getCharacter()) {
+      case 'd':
+      case ' ':
+        fieldScreenPresenter.switchFieldDisplay();
+        break;
+
+      case 's':
+        fieldScreenPresenter.setSortField();
+        break;
+
+      case 'q':
+        return fieldScreenPresenter.transitionToNextScreen();
+    }
+
+    return this;
+  }
+
+  public void showFieldScreen(String sortFieldHeader, List<Field> fields,
+    EnumMap<Field, Boolean> fieldDisplayMap, int currentPosition, int headerMaxLength,
+    int descriptionMaxLength, boolean moveMode) {
+    showScreenDescription(sortFieldHeader);
+
+    for (int i = 0; i < fields.size(); i ++) {
+      Field field = fields.get(i);
+      showField(i, field, fieldDisplayMap.get(field), i == currentPosition, headerMaxLength,
+        descriptionMaxLength, moveMode);
+    }
+  }
+
+  public void showScreenDescription(String sortKeyHeader) {
+    TerminalPrinter printer = getTerminalPrinter(SCREEN_DESCRIPTION_START_ROW);
+    printer.startBold().print("Fields Management").stopBold().endOfLine();
+    printer.print("Current Sort Field: ").startBold().print(sortKeyHeader).stopBold().endOfLine();
+    printer.print("Navigate with up/down, Right selects for move then <Enter> or Left commits,")
+      .endOfLine();
+    printer.print("'d' or <Space> toggles display, 's' sets sort. Use 'q' or <Esc> to end!")
+      .endOfLine();
+  }
+
+  public void showField(int pos, Field field, boolean display, boolean selected,
+    int fieldHeaderMaxLength, int fieldDescriptionMaxLength, boolean moveMode) {
+
+    String fieldHeader = String.format("%-" + fieldHeaderMaxLength + "s", field.getHeader());
+    String fieldDescription = String.format("%-" + fieldDescriptionMaxLength + "s",
+      field.getDescription());
+
+    int row = FIELD_START_ROW + pos;
+    TerminalPrinter printer = getTerminalPrinter(row);
+    if (selected) {
+      String prefix = display ? "* " : "  ";
+      if (moveMode) {
+        printer.print(prefix);
+
+        if (display) {
+          printer.startBold();
+        }
+
+        printer.startHighlight()
+          .printFormat("%s = %s", fieldHeader, fieldDescription).stopHighlight();
+
+        if (display) {
+          printer.stopBold();
+        }
+
+        printer.endOfLine();
+      } else {
+        printer.print(prefix);
+
+        if (display) {
+          printer.startBold();
+        }
+
+        printer.startHighlight().print(fieldHeader).stopHighlight()
+          .printFormat(" = %s", fieldDescription);
+
+        if (display) {
+          printer.stopBold();
+        }
+
+        printer.endOfLine();
+      }
+    } else {
+      if (display) {
+        printer.print("* ").startBold().printFormat("%s = %s", fieldHeader, fieldDescription)
+          .stopBold().endOfLine();
+      } else {
+        printer.printFormat("  %s = %s", fieldHeader, fieldDescription).endOfLine();
+      }
+    }
+  }
+}
```

Comment:
Random question: how does it do when screen size changes? Does it fill screen like htop? Mind putting up images on the JIRA so we can see how it looks?

Reply:
> how does it do when screen size changes?

hbtop can handle the screen size change. When screen size changes, it calculates the proper size of the top screen and change it.

> Does it fill screen like htop?

Yes.

> Mind putting up images on the JIRA so we can see how it looks?

Sure, will update the Jira.

Reply:
I see you posted the nice images of how it looks in your README posting on the issue. Thanks.